engine/classes/Elgg/Database/Plugins.php

```diff
@@ -154,14 +154,19 @@ public function getPath() {
 	 * @return void
 	 */
 	public function setBootPlugins($plugins) {
-		$this->boot_plugins = $plugins;
-		if (is_array($plugins)) {
-			foreach ($plugins as $plugin) {
-				if (!$plugin instanceof ElggPlugin || !$plugin->getID()) {
-					continue;
-				}
-				$this->cache->save($plugin->getID(), $plugin);
+		if (!is_array($plugins)) {
+			unset($this->boot_plugins);
+			return;
+		}
+		
+		foreach ($plugins as $plugin) {
+			$plugin_id = $plugin->getID();
```

Comment:
This will fatal if it's not a plugin instance

Reply:
check